### PR TITLE
[WIP] Add background layer button directive and example

### PR DESCRIPTION
**Pull request by adube**
This PR introduces the background layer button, which shows a background layer selector when clicked.

Please, wait until #1290 is merged before reviewing this one (it contains content from this PR too)

## Todo

 * [x] Wait for #1290 to be merged
 * [ ] Review

## Live demo

The image I have added doesn't seem to load properly in github.io.

 * https://adube.github.io/ngeo/desktop-bg-lyr-sel/examples/contribs/gmf/backgroundlayerbutton.html

### CODE REVIEW

File: contribs/gmf/src/directives/partials/backgroundlayerbutton.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  <gmf-backgroundlayerselector
+      gmf-backgroundlayerselector-map="::blbCtrl.map">
+  </gmf-backgroundlayerselector>
+</div>
+<a
+    href="#"
+    ng-click="blbCtrl.toggle();"
+    class="btn btn-default">
+  <img src="{{ blbCtrl.image }}" />
+</a>
```
Comment (sbrunner) [inline contribs/gmf/src/directives/partials/backgroundlayerbutton.html]: missing endline ...

**Comment by adube** [comment]
Endline added.  Anything else to review ?

**Comment by sbrunner** [comment]
Looks good for me, @fredj any comments?

**Comment by fredj** [comment]
Is it still in WIP? if not please update this pull request title and the commit messages (squash into one commit if possible)

**Comment by adube** [comment]
Yes, it is.  See in the description: "Wait for #1290 to be merged".  Its the follow-up of #1290.

**Comment by adube** [comment]
Ready for review.

**Comment by fredj** [comment]
This should fix the example image issue:
```diff
diff --git a/Makefile b/Makefile
index e45420c..2cc0e40 100644
--- a/Makefile
+++ b/Makefile
@@ -60,6 +60,7 @@ EXAMPLE_HOSTED_REQUIREMENTS = .build/examples-hosted/lib/ngeo.js \
        .build/examples-hosted/partials \
        .build/examples-hosted/data \
        .build/examples-hosted/contribs/gmf/data \
+       .build/examples-hosted/contribs/gmf/images \
        .build/examples-hosted/contribs/gmf/partials \
 
 # Git
@@ -382,6 +383,10 @@ dist/gmf.js.map: dist/gmf.js
        mkdir -p $@
        cp examples/data/* $@
 
+.build/examples-hosted/contribs/gmf/images: contribs/gmf/examples/images
+       mkdir -p $@
+       cp contribs/gmf/examples/images/* $@
+
 .build/examples-hosted/contribs/gmf/partials: contribs/gmf/examples/partials
        mkdir -p $@
        cp contribs/gmf/examples/partials/* $@
```

**Comment by pgiraud** [comment]
As discussed with @adube, this pull request needs rework.
I previously created a prototype using a simple dropdown and a menu in which the background selector was added.

**Comment by adube** [comment]
I'll simply abandon it and create an other one.  Sorry for the noise.

**Pull request closed without merging**